examples/README.md

```diff
@@ -0,0 +1,180 @@
+
+The second method is responsible for returning us the status of an assembly. Note that it
+extends ```Transloadit::Response::Assembly``` to give us access to convenience methods like
+```finished?```.
+
+### First example
+In the first example that gets played, we fetch an image from the cat api, optimize it
+using the Transloadit ```/image/optimize``` robot, and then store it in s3.
+
+There are only two steps:
+```
+optimize = transloadit_client.step('image', '/image/optimize', {
+  progressive: true,
+  use: ':original',
+  result: true
+})
+store = transloadit_client.step('store', '/s3/store', {
+  key: ENV.fetch('S3_ACCESS_KEY'),
+  secret: ENV.fetch('S3_SECRET_KEY'),
+  bucket: ENV.fetch('S3_BUCKET'),
+  bucket_region: ENV.fetch('S3_REGION'),
+  use: 'image'
+})
+ ```
+Again, we utilize environment variables to access our s3 credentials and pass them to 
+our assembly.
+
+The job is invoked by running the following:
+```
+assembly = transloadit_client.assembly(steps: [optimize, store])
+assembly.submit! open(file)
+```
+We pass the two steps we defined above and call ```open``` on the file passed in. This method
+assumes the file object passed in responds to ```open```.
+
+### Second example
+In the second example, we take a non-mp3 audio file, encode it as an mp3, add ID3 tags to it,
+and then store it in s3. There are many use cases for audio uploads, and adding ID3 tags
+provides the necessary metadata to display artist and track information in audio players
+such as iTunes.
+
+We have the following steps:
+
+```
+encode_mp3 = transloadit_client.step('mp3_encode', '/audio/encode', {
+  use: ':original',
+  preset: 'mp3',
+  ffmpeg_stack: 'v2.2.3',
+  result: true
+})
+write_metadata = transloadit_client.step('mp3', '/meta/write', {
+  use: 'mp3_encode',
+  ffmpeg_stack: 'v2.2.3',
+  result: true,
+  data_to_write: mp3_metadata
+})
+store = transloadit_client.step('store', '/s3/store', {
+  key: ENV.fetch('S3_ACCESS_KEY'),
+  secret: ENV.fetch('S3_SECRET_KEY'),
+  bucket: ENV.fetch('S3_BUCKET'),
+  bucket_region: ENV.fetch('S3_REGION'),
+  use: ['mp3']
+})
+```
+
+The first step simply uses the original file to create an mp3 version using the ```audio/encode```
+robot.
+
+The second step takes the first step as input, and adds the appropriate metadata using the ```meta/write```
+robot. In our simple example we set the track name to the name of the file using variable
+name substitution (see https://transloadit.com/docs/#assembly-variables), and set canned
+values for all other ID3 fields
+
+```
+def mp3_metadata
+  meta = { publisher: 'Transloadit', title: '${file.name}' }
+  meta[:album] = 'Transloadit Compilation'
+  meta[:artist] = 'Transloadit'
+  meta[:track] = '1/1'
+  meta
+end
+```
+
+Finally, we submit the assembly in the same way as the previous example:
+
+```
+assembly = transloadit_client.assembly(steps: [encode_mp3, write_metadata, store])
+assembly.submit! open(file)
+```
+
+### Third example
```

Comment:
Can you please add a new line after a new heading? Makes it easier to read the pure markdown without rendering :) Thanks!